controllers/admin_main.php

```diff
@@ -16,7 +18,17 @@ public function preAction()
     {
         parent::preAction();
 
-        $this->structure->set('page_title', Language::_('AdminMain.index.page_title', true));
+        // Load required models
+        $this->uses(['Clients', 'Companies', 'ModuleManager', 'Packages', 'Invoices', 'Services']);
+        $this->helpers(['Form', 'CurrencyFormat']);
+        $this->components(['Session']);
+
+        // Set the client info
+        if ($this->Session->read('blesta_client_id')) {
```

Comment:
This doesn't work in this context.  Instead you can do

`
        $client_id = ($this->get['client_id'] ?? ($this->get[0] ?? null));
        if ($client_id) {
            $this->client = $this->Clients->get($client_id);
        }
`